deployer/deployer.py

```diff
@@ -195,10 +201,209 @@ def deploy_version(self, version):
+        print("--> PARAMS: "
+              "version={version}, "
+              "health_check={health_check}, "
+              "health_check_port={health_check_port}, "
+              "remove_old={remove_old}".format(version=version,
+                                               health_check=health_check,
+                                               health_check_port=health_check_port,
+                                               remove_old=remove_old))
+
+        data = {
+            "deploymentOptions": {
+                "clusterName": self.app,
+                "notificationDestination": self.user_mail,
+                "steps": [
+                    {
+                        "type":                     "CreateAsg"
+                    }
+                ]
+            },
+            "asgOptions": {
+                "autoScalingGroupName": version,
+                "launchConfigurationName": None,
+                "minSize": self.min_instances,
+                "maxSize": self.max_instances,
+                "desiredCapacity": self.min_instances,
+                "defaultCooldown": 10,
+                "availabilityZones": ["eu-west-1c", "eu-west-1a", "eu-west-1b"],
+                "loadBalancerNames": self.elbs,
+                "healthCheckType": "EC2",
+                "healthCheckGracePeriod": 600,
+                "placementGroup": None,
+                "subnetPurpose": "external",
+                "terminationPolicies": ["Default"],
+                "tags": [],
+                "suspendedProcesses": []
+            },
+            "lcOptions": {
+                "launchConfigurationName": None,
+                "imageId": self.ami,
+                "keyName": self.key_name,
+                "securityGroups": self.security_group,
+                "userData": self.user_data,
+                "instanceType": self.instance_type,
+                "kernelId": "",
+                "ramdiskId": "",
+                "blockDeviceMappings": None,
+                "instanceMonitoringIsEnabled": False,
+                "instancePriceType": "ON_DEMAND",
+                "iamInstanceProfile": self.role,
+                "ebsOptimized": False,
+                "associatePublicIpAddress": True
+            }
+        }
+
+        r = self.request("deployment/start", json.dumps(data))
+
+        time.sleep(30)
+
+        self.resize_asg(asg_name=version)
+
+        good_deploy = self.wait_asg_ready(version=version,
+                                          health_check=health_check,
+                                          health_check_port=health_check_port)
+
+        if good_deploy:
+            if remove_old:
+                for asg_name in self.get_asg_names_in_cluster():
+                    if asg_name != version:
+                        print("Deleting ASG {0}".format(asg_name))
+                        self.delete_asg(asg_name)
+        else:
+            self.disable_asg(asg_name=version)
+
```

Comment:
@ferrangrau Hi, I think that you can have other method, this check remove_old, the code more or less:
```
def check_remove_old_version(self, old_version):
       if old_version:
                 for asg_name in self.get_asg_names_in_cluster():
                     if asg_name != version:
                         print("Deleting ASG {0}".format(asg_name))
                         self.delete_asg(asg_name)
```
and in method "deploy_version_without_eureka":
```
     if good_deploy:
        check_remove_old_version(old_version)
     else:
         ...
```